src/java_tools/junitrunner/java/com/google/testing/junit/runner/model/TestSuiteModel.java

```diff
@@ -91,7 +92,27 @@ private TestCaseNode getTestCase(Description description) {
 
   private TestNode getTest(Description description) {
     // The description shouldn't be null, but in the test runner code we avoid throwing exceptions.
-    return description == null ? null : testsMap.get(description);
+    if (description == null) {
+      return null;
+    }
+
+    TestNode test = testsMap.get(description);
+    if (test != null) {
+      return test;
+    }
+
+    final Optional<Description> matchingSuite = testsMap.keySet().stream()
```

Comment:
This still needs a comment explaining why this is being done.

Especially considering that now a "getTest" method can result in internal state being changed.